### PR TITLE
build: remove noop librdkafka build + minor refactor

### PR DESCRIPTION
**Pull request by joshuarli**
Update: I'm being silly; the existing Dockerfile build just downloads the confluent-kafka 1.2.0 wheel, so librdkafka 1.2.0 is already being used. This means the librdkafka build we're doing in the Dockerfile is a noop.

Outdated stuff:

[confluent-fakfa 1.2.0](https://pypi.org/project/confluent-kafka/1.2.0/#files) ships wheels with librdkafka pre-compiled in, so it's not necessary to build it. Although, this means, we don't get to control `LIBRDKAFKA_VERSION`. If we haven't had any issue in the past with this, I'd trust the confluent-kafka wheels to build with good librdkafka versions.

The bundled librdkafka seems to be consistent with confluent-akfak version: https://github.com/confluentinc/confluent-kafka-python/releases/tag/v1.2.0

So this effectively bumps librdkafka from 0.11.5 to 1.2.0.

That's... a large jump. @tkaemming

### CODE REVIEW

**Comment by joshuarli** [comment]
It also looks like there are py3.7 wheels shipped for 0.11.5: https://pypi.org/project/confluent-kafka/0.11.5/#files

So alteratively we could downgrade the python package. In theory they're using the same librdkafka, so as long as we don't use any changed python APIs, should be ok.

File: Makefile [hunk 1]
```diff
@@ -1,12 +1,4 @@
-UNAME := $(shell uname -s)
-
-ifeq ($(UNAME),Darwin)
-	librdkafka_cmd = install-librdkafka-homebrew
-else
-	librdkafka_cmd = install-librdkafka-src
-endif
-
-.PHONY: test install-python-dependencies install-librdkafka install-librdkafka-homebrew install-librdkafka-src-
```
Comment (tkaemming) [inline Makefile:9]: `install-librdkafka-src-`

🤔

**Review by tkaemming — APPROVED** [review]
Thanks for doing this. It's probably a good idea to work with @JTCunning to figure out when to merge/deploy this just to be extra careful.

File: Makefile (hunk 1 above)
Comment (joshuarli) [reply Makefile:9]: lol

**Pull request merged**